processing/src/main/java/io/druid/query/BySegmentQueryRunner.java

```diff
@@ -61,7 +62,11 @@ public BySegmentQueryRunner(
                   new BySegmentResultValueClass<T>(
                       results,
                       segmentIdentifier,
-                      query.getIntervals().get(0)
+                      Iterables.getOnlyElement(
+                          Iterables.getOnlyElement(query.getDataSources())
```

Comment:
For convenience, you had to add a lot of boilerplate, effectively emulating the behaviour which I suggested, because Iterables.getOnlyElement() throws exception if there are more than one element.